oscore-cf/src/main/java/org/eclipse/californium/oscore/ResponseDecryptor.java

```diff
@@ -78,16 +79,25 @@ public static Response decrypt(Response response) throws OSException {
 			throw new OSException(ErrorDescriptions.TOKEN_NULL);
 		}
 
-		byte[] plaintext = decryptAndDecode(enc, response, ctx, db.getSeqByToken(token));
-
-		DatagramReader reader = new DatagramReader(new ByteArrayInputStream(plaintext));
-
-		response = OptionJuggle.setRealCodeResponse(response,
-				CoAP.ResponseCode.valueOf(reader.read(CoAP.MessageFormat.CODE_BITS)));
-
-		// resets option so eOptions gets priority during parse
-		response.setOptions(EMPTY);
-		DataParser.parseOptionsAndPayload(reader, response);
+		//Check if parsing of response plaintext succeeds
+		byte[] plaintext;
```

Comment:
could the scope of plaintext be reduced by moving it into the "try" ?

Reply:
Yes I will change it accordingly.

Reply:
This has now been done in commit 7ffd314ade8fa58dc43f4facb1c65aff3d4180f2